python/UniverseWrapper.cpp

```diff
@@ -473,7 +473,7 @@ namespace FreeOrionPython {
             .def("hasTag",                      &UniverseObject::HasTag)
             .add_property("meters",             make_function(ObjectMeters,                 return_internal_reference<>()))
             .def("getMeter",                    make_function(ObjectGetMeter,               return_internal_reference<>()))
-            .def("dump",                        &UniverseObject::Dump,                      return_value_policy<return_by_value>(), "Returns string with debug information, use '0' as argument.")
+            .def("dump",                        +[](const UniverseObject& obj) {return obj.Dump(0); })
```

Comment:
Looks like it required for GCC, but fails at MSVC. Can it be rewritten to work with both compilers?

Reply:
Not sure, but could use an `#ifdef` if nothing else works

Reply:
````
diff --git a/python/UniverseWrapper.cpp b/python/UniverseWrapper.cpp
index a8b9c530e..2cdd5ac0e 100644
--- a/python/UniverseWrapper.cpp
+++ b/python/UniverseWrapper.cpp
@@ -51,6 +51,9 @@ namespace boost {
 namespace {
     void                    DumpObjects(const Universe& universe)
     { DebugLogger() << universe.Objects().Dump(); }
+    std::string             ObjectDump(const UniverseObject& obj) {
+        return obj.Dump(0);
+    }
 
     // We're returning the result of operator-> here so that python doesn't
     // need to deal with std::shared_ptr class.
@@ -212,7 +215,7 @@ namespace {
         }
         return results;
     }
-    boost::function<std::vector<int> (const ShipDesign&)> AttackStatsFunc =                 &AttackStatsP;
+    boost::function<std::vector<int> (const ShipDesign&)> AttackStatsFunc =                     &AttackStatsP;
 
     std::vector<ShipSlotType> HullSlots(const HullType& hull) {
         std::vector<ShipSlotType> retval;
@@ -473,7 +476,7 @@ namespace FreeOrionPython {
             .def("hasTag",                      &UniverseObject::HasTag)
             .add_property("meters",             make_function(ObjectMeters,                 return_internal_reference<>()))
             .def("getMeter",                    make_function(ObjectGetMeter,               return_internal_reference<>()))
-            .def("dump",                        +[](const UniverseObject& obj) {return obj.Dump(0); })
+            .def("dump",                        make_function(&ObjectDump,                  return_value_policy<return_by_value>()),    "Returns string with debug information.")
         ;
 
         ///////////////////
````

Reply:
Yep, it looks better.

I've tried to split `Dump` without arguments and `Dump` with offset to two different methods but it requires massive changes.

Reply:
Offset-parameterized `Dump` isn't needed for the Python API, I think...

Reply:
That's funny. IIRC, this did pass CI back when I initially commited this and the lambda expression builds fine both on MSVC2015 and MSVC2017 on my system.

Reply:
@Morlic-fo Can you try proposed changes?

Reply:
Not sure what you mean. I think I tested with the diff above before posting it...

Reply:
@geoffthemedio I mean if @Morlic-fo add it to this PR.